### PR TITLE
EASY-1049 Upgrade Scallop

### PR DESCRIPTION
**Pull request by jo-pol**
@DANS-KNAW/easy 

Upgraded scallop for the first component.

### CODE REVIEW

**Comment by rvanheest** [comment]
Can you comment on the use of `\u00A0`? Why do it this way and not just with spaces?

**Comment by jo-pol** [comment]
Ok, I'll rework the non-breaking space.

**Comment by janvanmansum** [comment]
Actually, if `\u00A0` is the unicode codepoint for non-breaking space, it would be better to use this explicit way than to use an actual non-breaking space, which looks just the same as a breaking space.

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/FileItemConf.scala [hunk 1]
```diff
@@ -56,7 +60,7 @@ class FileItemConf(args: Seq[String]) extends ScallopConf(args) {
       " for the file and the ancestor folders that don't yet exist in the dataset")(mayNotExist)
   val format = opt[String](
     name = "format", short = 'f',
-    descr = "dcterms property format, the mime type of the file",
+    descr = s"dcterms property format, the mime type of the file (default$NB_SPACE=${NB_SPACE}application/octet-stream)",
```
Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/FileItemConf.scala]: I was tempted to propose to add `.replaceAll("&nbsp;"," ")` to [this line](https://github.com/scallop/scallop/commit/88ae5d6d00ad39d717457438e9201d21e1e808ff#diff-c8d77f5821c15084b74aaeb7cbb822a6R22). I would expect a plain space in the output but would have to proof that with a new unit test. But first let us see if the original functionality gets restored, see [this issue](https://github.com/scallop/scallop/commit/88ae5d6d00ad39d717457438e9201d21e1e808ff#diff-c8d77f5821c15084b74aaeb7cbb822a6R22)

File: src/main/scala/nl/knaw/dans/easy/stage/fileitem/FileItemConf.scala [hunk 2]
```diff
@@ -29,6 +29,10 @@ class FileItemConf(args: Seq[String]) extends ScallopConf(args) {
 
   printedName = "easy-stage-file-item"
   version(s"$printedName v${Version()}")
+
+  editBuilder(_.setHelpWidth(100))
+  val NB_SPACE = "\u00A0" // &nbsp; won't work when calling for help on a console
```
Comment (janvanmansum) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/FileItemConf.scala]: @jo-pol : `&nbsp;` is an [html-entity](http://www.w3schools.com/html/html_entities.asp). The help is plain text, so it doesn't make sense to expect `&nbsp;` to be replaced by the corresponding character.

Comment (jo-pol) [inline src/main/scala/nl/knaw/dans/easy/stage/fileitem/FileItemConf.scala]: That was what I meant but original defaults have been restored in scallop-V1.0.1 so I expect we can get rid of this again.

File: README.md [hunk 3]
```diff
@@ -40,66 +40,52 @@ the command `easy-stage-file-item`. It executes step 3 to stage one or more file
...
+                                        KNOWN, RESTRICTED_REQUEST, RESTRICTED_GROUP, NONE] (default = NONE)
+     -c, --creator-role  <arg>          specifies the role of the file item creator; either one of [ARCHIVIST,
+                                        DEPOSITOR] (default = DEPOSITOR)
+     -l, --file-location  <arg>         The file to be staged (only required for copying in case of non-mendeley
+                                        use case)
+     -m, --is-mendeley                  Stage the dataset as a "mendeley dataset"
+         --owner-id  <arg>              specifies the id of the owner/creator of the file item (defaults to the one
+                                        configured in the application configuration file)
+     -v, --visible-to  <arg>            specifies the visibility of the file item; either one of [ANONYMOUS, KNOWN,
+                                        RESTRICTED_REQUEST, RESTRICTED_GROUP, NONE] (default = ANONYMOUS)
```
Comment (jo-pol) [inline README.md]: I don't like these square brackets, might confuse <strike>be dangerous in a</strike> markdown editors too. What about the following punctuation?

```
specifies the visibility of the file item; either one of: ANONYMOUS, KNOWN,
RESTRICTED_REQUEST, RESTRICTED_GROUP, NONE (default = ANONYMOUS)
```

Comment (rvanheest) [inline README.md]: Seems good to me, especially in capital. If we really want to have some kind of brackets, we can also use `{ }`...

Comment (jo-pol) [inline README.md]: The capitals are required by the java enum in our legacy code which should be matched.

Comment (janvanmansum) [inline README.md]: Language note: leave out the word "either"; [it suggests that there are two alternatives](http://www.mijnwoordenboek.nl/vertaal/EN/NL/either)

**Pull request merged**